### PR TITLE
Fixes #235 Enforced a frontend linting step to protect PRs

### LINKED ISSUE #235: Enforce frontend linting from GitHub action
### Overview
This will add restrictions to frontend pull requests to ensure consistency across the codebase

### Action Items
  - [x] include a lint step in the frontend CI GitHub action process
    - [x] lint step should just use `npm run lint`
  - [x] test with a PR and make sure it gets rejected if the frontend code fails linting

### PR DESCRIPTION
**Pull request by sellnat77**
Fixes #235

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1 @@
+*.json
```
Comment (brodly) [inline .eslintignore]: Do we want to ignore json?

**Comment by brodly** [comment]
This is fine. We might run into an annoying issue where the linter is enforcing accessibility. We can turn that down to warning or off for now and then add accessibility for MVP+

File: .eslintignore (hunk 1 above)
Comment (sellnat77) [reply .eslintignore]: We have massive json files in the `/data` folder
We can include them but the lint error is very very noisy

**Comment by sellnat77** [comment]
Gonna take me a bit to address the existing errors but ill post in the channel when this PR is primed and ready

File: .eslintignore (hunk 1 above)
Comment (brodly) [reply .eslintignore]: Im ok to ignore then

File: src/components/common/Dropdown.jsx [hunk 2]
```diff
@@ -3,31 +3,51 @@ import React from 'react';
...
         <span className="icon is-small">
-          <i className="fas fa-angle-down" aria-hidden="true"></i>
+          <i className="fas fa-angle-down" aria-hidden="true" />
         </span>
       </button>
     </div>
     <div className="dropdown-menu" id="dropdown-menu" role="menu">
       <div className="dropdown-content">
-        <a href="#" className="dropdown-item">
+        <button
```
Comment (sellnat77) [inline src/components/common/Dropdown.jsx]: This was probably the most strange change but seeing as how this isnt being used yet im sure we can figure out a lint-safe way to do this

Comment (brodly) [reply src/components/common/Dropdown.jsx]: pull latest dev, this has been fixed

**Pull request merged**